connectivity/perf/common/metrics.go

```diff
@@ -0,0 +1,168 @@
+		},
+		Unit:   "ops/s",
+		Labels: labels,
+	}
+}
+
+type ThroughputMetric struct {
+	Throughput float64 `json:"Throughput"` // Throughput in bytes/s
+}
+
+func (metric *ThroughputMetric) ToPerfData(labels map[string]string) DataItem {
+	return DataItem{
+		Data: map[string]float64{
+			"Throughput": metric.Throughput / 1000000,
+		},
+		Unit:   "Mb/s",
+		Labels: labels,
+	}
+}
+
+type PerfResult struct {
+	Timestamp             time.Time
+	Latency               *LatencyMetric
+	TransactionRateMetric *TransactionRateMetric
+	ThroughputMetric      *ThroughputMetric
+}
+
+type PerfTests struct {
+	Tool     string
+	Test     string
+	SameNode bool
+	Scenario string
+	Sample   int
+	Duration time.Duration
+}
+
+type PerfSummary struct {
+	PerfTest PerfTests
+	Result   PerfResult
+}
+
+// These two structures are borrowed from kubernetes/perf-tests:
+// https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/util/perftype.go
+// this is done in order to be compatible with perfdash
+type DataItem struct {
+	// Data is a map from bucket to real data point (e.g. "Perc90" -> 23.5). Notice
+	// that all data items with the same label combination should have the same buckets.
+	Data map[string]float64 `json:"data"`
+	// Unit is the data unit. Notice that all data items with the same label combination
+	// should have the same unit.
+	Unit string `json:"unit"`
+	// Labels is the labels of the data item.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// PerfData contains all data items generated in current test.
+type PerfData struct {
+	// Version is the version of the metrics. The metrics consumer could use the version
+	// to detect metrics version change and decide what version to support.
+	Version   string     `json:"version"`
+	DataItems []DataItem `json:"dataItems"`
+	// Labels is the labels of the dataset.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+func getLabelsForTest(summary PerfSummary, metric string) map[string]string {
+	node := "other-node"
+	if summary.PerfTest.SameNode {
+		node = "same-node"
+	}
+	return map[string]string{
+		"scenario":  summary.PerfTest.Scenario,
+		"node":      node,
+		"test_type": summary.PerfTest.Tool + "-" + summary.PerfTest.Test,
+		"metric":    metric,
+	}
+}
+
+func ExportPerfSummaries(summaries []PerfSummary, reporitDir string) {
+	perfData := []DataItem{}
+	for _, summary := range summaries {
+		if summary.Result.Latency != nil {
+			labels := getLabelsForTest(summary, "Latency")
+			perfData = append(perfData, summary.Result.Latency.ToPerfData(labels))
+		}
+		if summary.Result.TransactionRateMetric != nil {
+			labels := getLabelsForTest(summary, "TransactionRate")
+			perfData = append(perfData, summary.Result.TransactionRateMetric.ToPerfData(labels))
+		}
+		if summary.Result.ThroughputMetric != nil {
+			labels := getLabelsForTest(summary, "Throughput")
+			perfData = append(perfData, summary.Result.ThroughputMetric.ToPerfData(labels))
+		}
+	}
```

Comment:
I'd recommend creating an interface for each of these metrics, something like `DataItemable`:

```go
type DataItemable interface {
    ToPerfData() DataItem
}
```

Then adjust the `PerfResult` to have a `Metrics` field rather than individual fields for each metric:

```go
type PerfResult struct {
    Metrics []DataItemable
}
```

You won't need to do all these nil checks and the for loop would become a bit cleaner.

See the recommendation above for how to get around needing to call `getLabelsForTest` specifically for each metric type [1].

Reply:
For context, @marseel and I talked about this offline and determined that this could be done as a follow up.